docs/development/development.md

```diff
@@ -0,0 +1,76 @@
+
+</details>
+
+---
+
+### **What's the dependency evaluation flow in the `kfd.yaml` and how does furyctl download them?**
+
+<details>
+  <summary>Answer</summary>
+
+The `kfd.yaml` file, specific to the distribution downloaded by `furyctl`, contains the definitions of dependencies that the distribution relies on.
+
+Furyctl reads the version information embedded within the `kfd.yaml` file to determine which dependencies need to be fetched. These dependencies are typically other resources or libraries required for the successful deployment or operation of the current distribution.
+
+Once identified, `furyctl` downloads or references these dependencies from either a local or remote repository. This ensures that the distribution is fully equipped with all the necessary components for deployment, preventing version mismatches and compatibility issues. The process ensures the environment is set up with the correct versions and configurations.
+
+</details>
+
+---
+
+### **What's inside the Makefile?**
+
+<details>
+  <summary>Answer</summary>
+
+The important commands in the `Makefile` are:
+
+- **`make generate-go-models`**: This command generates Go code from the JSON schema files. The generated code defines the data models used in the codebase, providing a structured representation of the resources and configurations used by `furyctl`. It essentially converts the schema into Go structs, which are essential for interacting with the configuration data programmatically.
+
+- **`make generate-docs`**: This command generates Markdown documentation from the schema files. It extracts the necessary information from the schemas and formats it into human-readable documentation, helping developers and users understand how to configure and use the distribution and resources. This documentation serves as the primary reference for anyone interacting with `furyctl`.
+</details>
+
+---
+
+### **What's the purpose of a private schema, and what are the differences from the public schema?**
+
+<details>
+  <summary>Answer</summary>
+
+The `public` schema serves as the base schema, which is shared and visible to all users. It defines the core structure and expected properties for a particular resource or configuration.
+
+The `private` schema is a modified version of the public schema, typically used internally within `furyctl`. It includes additional fields or configurations that should not be exposed in the public configuration files (like `furyctl.yaml`), but are still necessary for certain internal operations or customizations within the codebase.
+
+A notable case where the private schema is used is with the `EKSCluster` resource. Here, a patch is applied to the public schema to add internal fields that are required for `furyctl` to function but are not intended for end-user modification. This separation ensures that sensitive or internal details remain private while maintaining flexibility for internal customization.
```

Comment:
Let's add that the private schema is autogenerated? How it's autogenerated and where are the additional fields taken from?

Reply:
On top of what @nutellinoit said, I'd add: how could I add a private field to a schema?